src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts

```diff
@@ -419,6 +444,22 @@ export module DotNet {
     return value;
   });
 
+  const byteArrayRefKey = '__byte[]';
+  attachReviver(function reviveByteArray(key: any, value: any, byteArrays: Uint8Array[] | null) {
+    if (value && typeof value === 'object' && value.hasOwnProperty(byteArrayRefKey) && byteArrays) {
+      const index = value[byteArrayRefKey];
+
+      if (index >= byteArrays.length || index < 0) {
+        throw new Error(`Byte array index '${index}' does not exist.`)
+      }
+
+      return byteArrays[value[byteArrayRefKey]];
+    }
+
+    // Unrecognized - let another reviver handle it
+    return value;
+  });
```

Comment:
Rather than us building an increasingly long list of JSON revivers, do you think it would be viable to coalesce all our built-in ones into a single function? This could hugely reduce the number of operations we're doing altogether, as they wouldn't all be chained into each other, and there would only need to be one top-level check for `value && typeof value === 'object'` etc.

If you feel this is beyond the scope of the PR that's totally fine, as we could change that later.